odl/operator/fn_ops.py

```diff
@@ -0,0 +1,406 @@
+        >>> ind = list()
+        >>> ind.append(range(2))
+        >>> ind.append([1,1])
+        >>> A = odl.operator.SamplingOperator(X, ind)
+        >>> A.adjoint(A(x))
+        uniform_discr([0.0, 0.0], [1.0, 1.0], (2, 2)).element([[0.0, 0.25], [0.0, 0.25]])
+        >>> A.adjoint(A(x))  # Out-of-place
+        uniform_discr([0.0, 0.0], [1.0, 1.0], (2, 2)).element([[0.0, 0.25], [0.0, 0.25]])
+
+        >>> X = odl.uniform_discr(min_pt=[-1, -1], max_pt=[1, 1], shape=[4, 3])
+        >>> x = X.element(range(1,X.size+1))
+        >>> ind = list()
+        >>> ind.append(range(3))
+        >>> ind.append(range(3))
+        >>> A = odl.operator.SamplingOperator(X, ind)
+        >>> A.adjoint(A(x)).inner(x) - A(x).inner(A(x))
+        -1.7763568394002505e-15
+        >>> A.adjoint(A(x))  # Out-of-place
+        -1.7763568394002505e-15
+        """
+
+        self.__indices = np.asarray(indices, dtype=int)
+        # Converts a single sequence to an array of integers
+        # and a sequence of arrays to a vertically stacked array
+        if self.indices.ndim > 1:
+            # Strides = increment in linear indices per axis
+            strides = np.concatenate((np.cumprod(range.shape[1:])[::-1], [1]))
+            self.__flat_indices = np.sum(self.indices * strides[:, None], axis=0)
+
+        self.__flat_indices_unique = np.unique(self.flat_indices)
+
+        domain = fn(self.flat_indices.size, dtype=range.dtype)
+        super().__init__(domain, range, linear=True)
+            
+    @property
+    def indices(self):
+        """indices where to sample the function."""
+        return self.__indices
+
+    @property
+    def flat_indices(self):
+        """flattened set of indices where to sample the function."""
+        return self.__flat_indices
+
+    @property
+    def flat_indices_unique(self):
+        """unique set of indices where to sample the function."""
+        return self.__flat_indices_unique
+
+    def _call(self, x, out=None):
+        """sum all values if indices are given multiple times."""        
+        y = np.bincount(self.flat_indices, weights=x, minlength=self.range.size)
+            
+        if out is None:
+            out = self.range.element(y)
+        else:
+            out[:] = y
+
+        return out
+
+    @property
+    def adjoint(self):
+        """Adjoint of the sampling is 'sampling'.
+
+        Returns
+        -------
+        adjoint : `SamplingOperator`
+        """
+        return SamplingOperator(self.domain, self.indices)
+
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        return '{}({!r}, {!r})'.format(self.__class__.__name__,
+                                       self.domain, self.indices)
+                
+
+    def __str__(self):
+        """Return ``str(self)``."""
+        return '{} * I'.format(self.indices)
+
+
+class VectoriseOperator(Operator):
+
+    """Operator that reshapes the object as a column vector.
+
+        ``VectoriseOperator(f) == f[:]``
```

Comment:
This looks like matlab syntax to me, since in python this is a no-op.

I suspect you mean

```
VectoriseOperator(f) == np.ravel(f)
```

or something like it?